sacred/config/signature.py

```diff
@@ -67,36 +70,45 @@ def __unicode__(self):
     def __repr__(self):
         return "<Signature at 0x{1:x} for '{0}'>".format(self.name, id(self))
 
-    def _assert_no_unexpected_args(self, args):
-        if not self.vararg_name and len(args) > len(self.arguments):
-            unexpected_args = args[len(self.arguments):]
+    def _get_expected_args(self, bound):
+        if bound:
+            # When called as instance method, the instance ('self') will be
+            # passed as first argument automatically, so the first argument
+            # should be excluded from the signature during this invocation.
+            return self.arguments[1:]
```

Comment:
For clarity, this is the essential line of this PR, excluding the first item from `self.arguments`. All other changes are just to pass this value around.